Ports/RISC-V/RV32/GCC/os_cpu_a.S

```diff
@@ -153,19 +169,92 @@ OSStartHighRdy:
 #              the result of an interrupt.  This function triggers a synchronous software interrupt by
 #              writing into the MSIP register
 #********************************************************************************************************
+OSIntCtxSw:
+    ret
 
 OSCtxSw:
-OSIntCtxSw:
-# MIE_MSIE -- enable software interrupt bit
-    li     t0, RISCV_MIE_MSIE
-    csrrs  zero, mie, t0
-
-# This will trigger a synchronous software interrupt; PRCI->MSIP[0] = 0x01;
-    li     t0, RISCV_PRCI_BASE_ADDR
-    li     t1, 0x1
-    sw     t1, 0x0(t0)
+    ecall
     ret
 
+.align 8
+ucos_intr_exception_handler:
+    # save regs to current SP
+    addi sp, sp, -4*32
+
+    sw     ra,   0 * 4(sp)
+    sw     t0,   4 * 4(sp)
+    sw     t1,   5 * 4(sp)
+    sw     t2,   6 * 4(sp)
+    sw     s0,   7 * 4(sp)
+    sw     s1,   8 * 4(sp)
+    sw     a0,   9 * 4(sp)
+    sw     a1,  10 * 4(sp)
+    sw     a2,  11 * 4(sp)
+    sw     a3,  12 * 4(sp)
+    sw     a4,  13 * 4(sp)
+    sw     a5,  14 * 4(sp)
+    sw     a6,  15 * 4(sp)
+    sw     a7,  16 * 4(sp)
+    sw     s2,  17 * 4(sp)
+    sw     s3,  18 * 4(sp)
+    sw     s4,  19 * 4(sp)
+    sw     s5,  20 * 4(sp)
+    sw     s6,  21 * 4(sp)
+    sw     s7,  22 * 4(sp)
+    sw     s8,  23 * 4(sp)
+    sw     s9,  24 * 4(sp)
+    sw     s10, 25 * 4(sp)
+    sw     s11, 26 * 4(sp)
+    sw     t3,  27 * 4(sp)
+    sw     t4,  28 * 4(sp)
+    sw     t5,  29 * 4(sp)
+    sw     t6,  30 * 4(sp)
+
+
+ 
+    li t1, ARCH_CPU_MCAUSE_CAUSE_MASK
```

Comment:
Put comment here, basically if its a interrupt we dont add +4 and if its a exception we add +4 to mepc